### PR TITLE
mysql-compatibility: add doc for a UPDATE incompatibility case

### PR DESCRIPTION
**Pull request by tiancaiamao**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/7094
- Other reference link(s):

https://github.com/pingcap/tidb/issues/25645

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@tiancaiamao Please involve one or two tech reviewers. Thanks~

**Comment by tiancaiamao** [comment]
> @tiancaiamao Please involve one or two tech reviewers. Thanks~

OK, I'll address comment from @morgo and after I update the PR, I'll do it.

File: mysql-compatibility.md [hunk 1]
```diff
@@ -123,6 +123,10 @@ TiDB 中的[信息统计](/statistics.md#手动收集)与 MySQL 中的有所不
 
 详情参见 [`SELECT`](/sql-statements/sql-statement-select.md)。
 
+### `UPDATE` 语句
+
+详情参见 [`UPDATE`](/sql-statements/sql-statement-update.md) statement reference.
```
Comment (shichun-0415) [inline mysql-compatibility.md:128]:
```suggestion
详情参见 [`UPDATE`](/sql-statements/sql-statement-update.md).
```

**Review by shichun-0415 — APPROVED** [review]

File: sql-statements/sql-statement-update.md [hunk 2]
```diff
@@ -104,7 +104,17 @@ SELECT * FROM t1;
...
-`UPDATE` 语句与 MySQL 完全兼容。如发现任何兼容性差异，请在 GitHub 上提交 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
+在计算表达式中的列时，TiDB 总使用原始的值。例如：
+
+```sql
+CREATE TABLE t (a int, b int);
+INSERT INTO t VALUES (1,2);
+UPDATE t SET a = a+1,b=a;
+```
+
+在 MySQL 中，`b` 列的值会被更新成 2，因为它是设置成了 `a`，而 `a`（最初是 1）在同一条语句中被更新成了 2。
```
Comment (TomShawn) [inline sql-statements/sql-statement-update.md:115]:
```suggestion
在 MySQL 中，`b` 列的值会被更新成 2，因为 `b` 列被设置为与 `a` 列相同，而 `a`（最初是 1）在同一条语句中被更新成了 2。
```

File: sql-statements/sql-statement-update.md [hunk 3]
```diff
@@ -104,7 +104,17 @@ SELECT * FROM t1;
...
+
+```sql
+CREATE TABLE t (a int, b int);
+INSERT INTO t VALUES (1,2);
+UPDATE t SET a = a+1,b=a;
+```
+
+在 MySQL 中，`b` 列的值会被更新成 2，因为 `b` 列被设置为与 `a` 列相同，而 `a`（最初是 1）在同一条语句中被更新成了 2。
+
+TiDB 遵守标准的 SQL 行为，这里会将会 `b` 更新成 1。
```
Comment (TomShawn) [inline sql-statements/sql-statement-update.md:117]:
```suggestion
TiDB 遵守标准的 SQL 行为，这里将 `b` 列值更新成 1。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**